feat(OSC onboarding): New OSC form and validator

gustavlrsn opened the pull request:
Resolves https://github.com/opencollective/opencollective/issues/5973 & https://github.com/opencollective/opencollective/issues/5976
Requires https://github.com/opencollective/opencollective-api/pull/8050

# Description
Implements the updated OSC application form.
Also surfaces the OSC validation information in the pending application in the host dashboard.

# Screenshots
<img width="1344" alt="image" src="https://user-images.githubusercontent.com/1552194/195328763-95d4e488-4a95-42f3-af17-b4ae6a69af19.png">
<img width="1043" alt="image" src="https://user-images.githubusercontent.com/1552194/195328941-18d0e107-02d7-4486-b856-46fac5e07b91.png">

kewitz commented on components/osc-host-application/ApplicationForm.js:
```diff
@@ -293,46 +370,43 @@ const ApplicationForm = ({
                       {!LoggedInUser && (
-                        <Flex flexDirection={['column', 'row']}>
-                          <Box mr={[null, 3]}>
+                        <Grid gridTemplateColumns={['1fr', 'repeat(2, minmax(0, 1fr))']} gridGap={3} py={2}>
```
Working great on mobile.

kewitz reviewed (approved):
Solid. Not sure if needed, but should we use images with higher resolution?

gustavlrsn commented:
> Solid. Not sure if needed, but should we use images with higher resolution?

Thanks. Good point, higher res images would make sense especially since we're using the Next.js Image component that will automatically optimize them. I'll fix that and then merge.

gustavlrsn commented:
Added some small fixes:
- higher res images for the form
- update pick repo FAQ to use new style through props (`withNewButtons`, `withBorderLeft`)
- move invited admin count (e.g `(1/2`) from sitting below to be on same row as title (`Added administrators`), same as other forms
- added `Show Less` and `Expand` as internationalized strings

Nothing major, so merging this now.
